Source/WTF/Scripts/analyze-spi.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+set -e
+
+program="${SRCROOT}/../../../${WK_ADDITIONAL_SCRIPTS_DIR}/analyze-spi"
+[ -f "${program}" ] || program="${SRCROOT}/../../Tools/Scripts/analyze-spi"
+program=$(realpath "${program}")
+
+if [[ "${WK_ANALYZE_SPI}" == YES && -f "${program}" ]]; then
+    mkdir -p "${OBJROOT}/WebKitSDKDBs"
+    (set -x && "${program}" \
+     --sdkdb-dir "${WK_SDKDB_DIR}/${WK_SDKDB_NAME}" \
+     --sdkdb-cache "${OBJROOT}/WebKitSDKDBs/${WK_SDKDB_NAME}.sqlite3" \
+     --sdk-dir "${SDKROOT}" --arch-name "${ARCHS%% *}" \
```

Comment:
Can you explain what the `${ARCHS%% *}` does? I can't find info other than that it's a greedy substitution modifier.

Reply:
It's a bash-ism to get the first whitespace-separate word from a variable. So what's happening here is that if a build is multi-arch, we take the first architecture from the list and use that for analysis.

…which is perhaps a dangerous corner to cut. For example, our macOS CI happened to pass `ARCHS="x86_64 arm64"` we'd only ever check the legacy architecture.

Reply:
Latest revision checks all architectures, which will be a tiny bit more work to do for CI builders, but will have no impact at desk.